bittensor/_cli/cli_impl.py

```diff
@@ -256,15 +256,23 @@ def set_weights( self ):
     def _get_hotkey_wallets_for_wallet( wallet ):
         hotkey_wallets = []
         hotkeys_path = wallet.path + '/' + wallet.name + '/hotkeys'
-        hotkey_files = next(os.walk(os.path.expanduser(hotkeys_path)))[2]
+        try:
+            hotkey_files = next(os.walk(os.path.expanduser(hotkeys_path)))[2]
+        except StopIteration:
+            hotkey_files = []
         for hotkey_file_name in hotkey_files:
             hotkey_wallets.append( bittensor.wallet( path = wallet.path, name = wallet.name, hotkey = hotkey_file_name ))
         return hotkey_wallets
 
     def list(self):
         r""" Lists wallets.
         """
```

Comment:
This comment isn't addressed at this PR, but this method is in dire need of some TLC and refactoring. We have 4 levels of indentation with static indexing into some lists, a bunch of bare try/except statements. I think a wallet named `"?"` could mean it's encrypted but it could also mean an unknown error was raised at some point and silently executed.